docs/manuals/gui/extension/dynamic_element/index.md

```diff
@@ -0,0 +1,329 @@
+  complete description of the `tsconfig.json` file content:
+
+    ```json hl_lines="4 10"
+    {
+      "compilerOptions": {
+        "target": "es5",
+        "outDir": "./dist/",
+        "module": "esnext",
+        "moduleResolution": "node",
+        "jsx": "react-jsx",
+        "allowJs": true,
+      },
+      "include": [ "src" ]
+    }
+    ```
+    Note that you must make sure that the directory where TypeScript files are located
+    (the "include" property) and the output directory (the "outDir" property) are set
+    according to your project structure.
+
+- `webpack` must be configured so it can transpile TypeScript code.<br/>
+  The two essential settings in the `webpack.config.js` file are shown here:
+
+    ```json
+    resolve: { extensions: [".ts", ".tsx"] },
+    module: {
+      rules: [
+        {
+          test: /\.tsx?$/,
+          use: "ts-loader",
+          exclude: /node_modules/
+        }
+      ]
+    },
+    ```
+
+   The settings of the *resolve* and *rules* properties make `webpack` look for TypeScript
+   files (`.ts` and `.tsx`) and process those files using the "ts-loader" TypeScript
+   compiler.
+
+### Resolve Taipy GUI dependency
+
+As you can see in the implementation code for React components, we depend on a
+JavaScript library that Taipy GUI provides.<br/>
+In the source code for the components, you will find directives similar to the
+following:
+```js
+import { ... } from "taipy-gui";
+```
+
+The "taipy-gui" module is provided by Taipy GUI, where it has been installed.<br/>
+You can query where Taipy GUI was installed by issuing the command:
+```
+pip show taipy-gui
+```
+
+To simplify the understanding of the rest of this section, let's assume that you store
+the directory path returned by this command in the environment variable 'TAIPY_GUI_DIR'.<br/>
+
+To install the Taipy GUI JavaScript module, you need to run the following command:
+```
+npm i $TAIPY_GUI_DIR/taipy/gui/webapp
+```
+
+Note that in the `package.json` file that is installed in Taipy GUI (in
+`$TAIPY_GUI_DIR/taipy/doc/extension/example_library/front-end`), there is an entry
+called "install" in the "scripts" section that invokes an NPM script to perform
+this task.
+
+### Bundle the JavaScript code
+
+`webpack` must be configured to indicate where the front-end JavsScript is produced.<br/>
+This is done by setting the *output* property in `webpack.config.js`:
+```js
+    output: {
+      filename: "extensionLibrary.js",
+      path: path.resolve(__dirname, "dist"),
+      library: {
+        // Camel case transformation of the library name
+        name: "MyExtensionLibrary",
+        type: "umd"
+      },
+    },
+```
+This indicates that all the source files should be bundled in a single module file
+called 'extensionLibrary.js', located in the 'dist' directory.
+
+When the different configuration files are up-to-date, building the extension library
+bundle is just a matter of running:
+
```

Comment:
Mention `externals: {"taipy-gui": "TaipyGui"},` 
And 
```
plugins: [
            new webpack.DllReferencePlugin({
```?
This is important as it will prevent webpack from bundling react & co...

Reply:
VERY correct!